### DISCUSSION TITLE
SFTP over GVFS unreliable

### DISCUSSION THREAD

**Discussion by wojnilowicz** [doublecmd/doublecmd#1650] [General]
When I enter sftp://wojnilowicz@fedorapeople.org/home/fedora/wojnilowicz/ in a way mentioned at
https://github.com/doublecmd/doublecmd/discussions/1133#discussioncomment-8566770
it frequently hangs at listing files with following line in the terminal `Creating TGioFileSource`
After Ctrl+C in the terminal I get
```
PollThread: RemovePoll 23
PollThread: AddPoll 23
Finished Double Commander
Destroying main form
Drive watcher finished
Waiting for FileSystemWatcher thread
FileSystemWatcher thread finished
Main form destroyed
Waiting for workers thread 000000000393A098
Finalized workers thread   000000000393A098
Waiting for workers thread 0000000003939608
```
If it succeeds and the connection is unused for a little while, then it hangs still at listing files. Restarting DC doesn't help. It's not reliable.

Accessing my space as mentioned at https://github.com/doublecmd/doublecmd/issues/1604#issue-2277427165 works more reliably though.  If the connection is unused for a little while then I get `Error: [6], SSH ERROR Timed out waiting on socket` and I have to reconnect.

Does GVFS method work as expected or is it a bug?

I use a SSH key but without ssh-agent.
```
Double Commander
Version: 1.1.12 gamma
Revision: 436
Commit: c8f17e36a
Build date: 2024/04/08
Lazarus: 2.2.6.0
FPC: 3.2.2
Platform: x86_64-Linux-qt5
OS version: Fedora release 39 (Thirty Nine)

Widgetset library: Qt 5.15.12, libQt5Pas 5.6.1
```

**Comment by alexx2000** [comment]
I've never encountered anything like this.
